src/main/java/com/bookshop/controller/UserController.java

```diff
@@ -0,0 +1,112 @@
+package com.bookshop.controller;
+
+import com.bookshop.dto.user.UserRegistrationRoleRequestDto;
+import com.bookshop.dto.user.UserRegistrationRoleResponseDto;
+import com.bookshop.exception.RegistrationException;
+import com.bookshop.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/user")
+@RequiredArgsConstructor
+@RestController
+public class UserController {
+    private static final String INCORRECT_DATA =
```

Comment:
You sure that you need to create this constant in each controller? Maybe it's better to have some centralised place for it